src/aws/osml/tile_server/utils/log_tools.py

```diff
@@ -0,0 +1,50 @@
+#  Copyright 2023-2024 Amazon.com, Inc. or its affiliates.
+
+import threading
+from logging import Filter, Formatter, Logger, LogRecord
+from typing import List, Optional
+
+_LOG_CONTEXT = threading.local()
+
+
+class ThreadingLocalContextFilter(Filter):
+    """
+    This is a filter that injects contextual information into the log message. The contextual information is
+    set using the static methods of this class.
+    """
+
+    def __init__(self, attribute_names: List[str]):
+        super().__init__()
+        self.attribute_names = attribute_names
+
+    def filter(self, record: LogRecord) -> bool:
+        """
+        This method is called for each log record. It injects the contextual information into the log record.
+
+        :param record: the log record to filter
+        :return: True, this filter does not exclude information from the log
+        """
+        for attribute_name in self.attribute_names:
+            setattr(record, attribute_name, getattr(_LOG_CONTEXT, attribute_name, None))
+        return True
+
+    @staticmethod
+    def set_context(context: Optional[dict]):
```

Comment:
Missing return type and update docstring to reflect on the return type.

Reply:
fixed